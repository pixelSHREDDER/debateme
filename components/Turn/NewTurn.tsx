'use client'

import addTurn from '@/actions/add-turn'
import { TDebate } from '@/lib/prisma-types'
import { useFormState } from 'react-dom'
import { useRef, useState } from 'react'
import { useUser } from '@auth0/nextjs-auth0/client'
import Editor from '@/components/Editor/Editor'
import FormSubmitButton from '@/components/Forms/FormSubmitButton'
import FormAlert from '@/components/Forms/FormAlert'
import { Button, Fieldset, Flex, Modal, Stack, Text } from '@mantine/core'
import Quotes from '@/components/Quotes/Quotes'
import { useDisclosure } from '@mantine/hooks'
import useCustomMarks from '@/hooks/useCustomMarks'

const SUBMIT_WARNING = 'Remember: turns can\'t be edited, and you can\'t submit another turn until your opponent finishes theirs.'

const initialState = {
  message: '',
}

export default function NewTurn({ debate, onSubmit }: { debate: TDebate, onSubmit: Function }) {
  const [isProofread, setIsProofread] = useState<boolean>(false)
  const [state, formAction] = useFormState(addTurn, initialState)
  const bodyRef = useRef<HTMLInputElement>(null)
  const { user } = useUser()
  const [opened, { open, close }] = useDisclosure(false)
  const { stripCustomMarks } = useCustomMarks()

  const onUpdate = (editorHtml?: string) => {
    if (!!editorHtml && bodyRef.current) bodyRef.current.value = stripCustomMarks(editorHtml)
  }

  const onProofread = () => { if (!isProofread) setIsProofread(true) }

  const onTurnSubmit = (payload: FormData) => {
    formAction(payload)
    if (onSubmit) {
      onSubmit()
    }
  }

  if (!user || !user.sub) {
    return ''
  }

  return (
    <form action={onTurnSubmit} id="new-turn-form" data-testid="new-turn-form">
      <Fieldset mt={30} mx={0} radius="lg" variant="filled" legend="It's your turn!">
        <input
          type="hidden"
          id="debateId"
          name="debateId"
          value={debate.id} />
        <input
          type="hidden"
          id="isCreator"
          name="isCreator"
          value={user.sub === debate.creatorSub ? 'true' : 'false'} />
        <input
          type="hidden"
          id="userSub"
          name="userSub"
          value={user.sub} />
        <input
          ref={bodyRef}
          type="hidden"
          id="bodyString"
          name="bodyString" />
        {/*<InputLabel htmlFor="body">New turn text</InputLabel>*/}
        <Editor
          id="body"
          onProofread={onProofread}
          onUpdate={onUpdate}
          required="true"
          testid="new-turn-editor">
        </Editor>
        {state?.message &&
          <FormAlert
            message={state.message}
            title="Something went wrong"
            type={2} />
        }
        <Flex align="center" gap={20} justify="flex-start">
          <Quotes />
          {!!isProofread &&
            <Button flex="1 0 auto" onClick={open}>Finish Your Turn</Button>
          }
        </Flex>
      </Fieldset>
      <Modal opened={opened} onClose={close} centered title="All set?">
        <Stack>
          <Text>{SUBMIT_WARNING}</Text>
          <Flex gap="sm">
            <FormSubmitButton form="new-turn-form" label="I'm All Done" />
            <Button variant="outline" onClick={close}>Not Yet</Button>
          </Flex>
        </Stack>
      </Modal>
    </form>
  )
}
