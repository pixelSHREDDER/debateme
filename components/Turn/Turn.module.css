.card {
  background-color: light-dark(var(--mantine-color-marble-2), var(--mantine-color-marble-light));
  margin-inline: min(40px, 3rem) 0;
  border-radius: var(--mantine-radius-lg) var(--mantine-radius-sm);
  box-shadow: 3px 5px 4px light-dark(rgba(0,0,0,0.6), rgba(255,255,255,0.3));
  overflow: visible;

  mark {
    background-color: light-dark(rgba(255, 255, 0, 1), rgba(255, 255, 0, 0.6));
  }
}

.opponentCard {
  background-color: light-dark(var(--mantine-color-purple-1), var(--mantine-color-purple-light));
  margin-inline: 0 min(40px, 3rem);
  border-radius: var(--mantine-radius-sm) var(--mantine-radius-lg);
}
