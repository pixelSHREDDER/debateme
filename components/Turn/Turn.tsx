'use client'

export default function Turn({ children }: { children: React.ReactNode }) {
  return <li>{children}</li>
}