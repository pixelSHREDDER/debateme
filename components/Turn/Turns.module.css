.list {
  display: grid;
  gap: 2em;
  padding-inline-start: 0;
}