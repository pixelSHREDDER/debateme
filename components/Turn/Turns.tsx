'use client'

import { Debate, Turn } from "@prisma/client"
//import React, { Suspense } from "react"

export default function Turns(props: {debate: Debate}) {
  /*const turns = React.useMemo(() => {
    !!props.debate.turn ? JSON.parse(props.debate.turn) : []
  }, [props.debate])

  return (
    <Suspense>
      {turns.map((turn: Turn) => <li>{turn.body}</li>)}
    </Suspense>
  )*/
  return null
}