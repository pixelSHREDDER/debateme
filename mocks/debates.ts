import { TDebate } from '@/lib/prisma-types'
import { DebateStatus } from '@prisma/client'
import { createTurns } from './turns'

export const createDebates: () => TDebate[] = () => {
  const createdAtDate = new Date('2024-04-13T12:24:02Z')
  const cooldownDate = new Date()
  const createdTurns = createTurns()

  return [
    {
      id: 1,
      topic: 'Your turn',
      createdAt: createdAtDate,
      creatorSub: 'auth0|1234567890abcdefghij1234',
      opponentSub: 'auth0|abcdefghij1234567890abcd',
      cooldownMins: 60,
      turn: createdTurns,
      status: DebateStatus.CreatorTurn,
    },
    {
      id: 2,
      topic: 'Opponent\'s Turn',
      createdAt: createdAtDate,
      creatorSub: 'auth0|1234567890abcdefghij1234',
      opponentSub: 'auth0|abcdefghij1234567890abcd',
      cooldownMins: 60,
      status: DebateStatus.OpponentTurn,
      turn: [
        createdTurns[0],
      ],
    },
    {
      id: 3,
      topic: 'Wait for Cooldown',
      createdAt: createdAtDate,
      creatorSub: 'auth0|1234567890abcdefghij1234',
      opponentSub: 'auth0|abcdefghij1234567890abcd',
      cooldownMins: 60,
      status: DebateStatus.CreatorCooldown,
      turn: [
        createdTurns[0],
        {
          ...createdTurns[1],
          createdAt: cooldownDate,
        },
      ],
    },
    {
      id: 4,
      topic: 'No opponent',
      createdAt: createdAtDate,
      creatorSub: 'auth0|1234567890abcdefghij1234',
      opponentSub: null,
      cooldownMins: 60,
      status: DebateStatus.NoOpponent,
    },
    {
      id: 5,
      topic: 'You\'re Invited to join',
      createdAt: createdAtDate,
      creatorSub: 'auth0|abcdefghij1234567890abcd',
      opponentSub: null,
      cooldownMins: 60,
      status: DebateStatus.NoOpponent,
    },
  ] as const
}
