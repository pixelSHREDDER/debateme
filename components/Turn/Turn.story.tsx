import type { Meta, StoryObj } from '@storybook/react'
import { createTurns } from '@/mocks/turns'
import Turn from './Turn'

const meta: Meta<typeof Turn> = {
  title: 'Turn',
  component: Turn,
}

const turns = createTurns()

export default meta

type Story = StoryObj<typeof Turn>

export const YourTurn: Story = {
  args: {
    body: turns[0].body || '',
    createdAt: turns[0].createdAt,
    isOpponent: false,
  },
}

export const OpponentsTurn: Story = {
  args: {
    body: turns[1].body || '',
    createdAt: turns[1].createdAt,
    isOpponent: true,
  },
}
