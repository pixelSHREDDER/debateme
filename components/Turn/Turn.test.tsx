import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest'
import { render } from '@testing-library/react'
import ProviderWrapper from '@/tests/helpers/ProviderWrapper'
import { createTurns } from '@/mocks/turns'
import Turn from './Turn'

const turns = createTurns()

describe('Turn component', () => {
  beforeEach(() => {
    vi.useFakeTimers()
  })

  afterEach(() => {
    vi.useRealTimers()
  })

  describe('Your turn', () => {
    test('renders correctly', () => {
      const { container } = render(
        <Turn createdAt={turns[0].createdAt} isOpponent={false}>
          <div dangerouslySetInnerHTML={{ __html: turns[0].body }} />
        </Turn>,
        { wrapper: ProviderWrapper },
      )
      expect(container).toMatchSnapshot()
    })
  })

  describe('Opponent\'s turn', () => {
    test('renders correctly', () => {
      const { container } = render(
        <Turn createdAt={turns[0].createdAt} isOpponent={true}>
          <div dangerouslySetInnerHTML={{ __html: turns[0].body }} />
        </Turn>,
        { wrapper: ProviderWrapper },
    )
      expect(container).toMatchSnapshot()
    })
  })
})
