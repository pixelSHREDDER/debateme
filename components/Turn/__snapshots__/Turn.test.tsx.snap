// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Turn component > Opponent's turn > renders correctly 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: mocked, sans-serif;--mantine-font-family-headings: mocked, sans-serif;--mantine-primary-color-filled: var(--mantine-color-marble-filled);--mantine-primary-color-filled-hover: var(--mantine-color-marble-filled-hover);--mantine-primary-color-light: var(--mantine-color-marble-light);--mantine-primary-color-light-hover: var(--mantine-color-marble-light-hover);--mantine-primary-color-light-color: var(--mantine-color-marble-light-color);--mantine-primary-color-0: var(--mantine-color-marble-0);--mantine-primary-color-1: var(--mantine-color-marble-1);--mantine-primary-color-2: var(--mantine-color-marble-2);--mantine-primary-color-3: var(--mantine-color-marble-3);--mantine-primary-color-4: var(--mantine-color-marble-4);--mantine-primary-color-5: var(--mantine-color-marble-5);--mantine-primary-color-6: var(--mantine-color-marble-6);--mantine-primary-color-7: var(--mantine-color-marble-7);--mantine-primary-color-8: var(--mantine-color-marble-8);--mantine-primary-color-9: var(--mantine-color-marble-9);--mantine-color-gray-0: #f0f3fd;--mantine-color-gray-1: #e0e3ee;--mantine-color-gray-2: #bfc4d9;--mantine-color-gray-3: #9ca4c5;--mantine-color-gray-4: #7e88b3;--mantine-color-gray-5: #6b77a9;--mantine-color-gray-6: #616ea5;--mantine-color-gray-7: #515d91;--mantine-color-gray-8: #475283;--mantine-color-gray-9: #3a4775;--mantine-color-marble-0: #faf7ea;--mantine-color-marble-1: #efecde;--mantine-color-marble-2: #dbd7bf;--mantine-color-marble-3: #c8c29d;--mantine-color-marble-4: #b7af81;--mantine-color-marble-5: #aca36e;--mantine-color-marble-6: #a79d62;--mantine-color-marble-7: #918951;--mantine-color-marble-8: #817945;--mantine-color-marble-9: #706836;--mantine-color-purple-0: #f6eeff;--mantine-color-purple-1: #e7daf7;--mantine-color-purple-2: #cab1ea;--mantine-color-purple-3: #ad86dd;--mantine-color-purple-4: #9562d2;--mantine-color-purple-5: #854bcb;--mantine-color-purple-6: #7d3ec9;--mantine-color-purple-7: #6b31b2;--mantine-color-purple-8: #5f2aa0;--mantine-color-purple-9: #52228d;}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-marble-4);--mantine-color-gray-light: rgba(97, 110, 165, 0.15);--mantine-color-gray-light-hover: rgba(97, 110, 165, 0.2);--mantine-color-gray-outline-hover: rgba(126, 136, 179, 0.05);--mantine-color-marble-text: var(--mantine-color-marble-4);--mantine-color-marble-filled: var(--mantine-color-marble-8);--mantine-color-marble-filled-hover: var(--mantine-color-marble-9);--mantine-color-marble-light: rgba(167, 157, 98, 0.15);--mantine-color-marble-light-hover: rgba(167, 157, 98, 0.2);--mantine-color-marble-light-color: var(--mantine-color-marble-3);--mantine-color-marble-outline: var(--mantine-color-marble-4);--mantine-color-marble-outline-hover: rgba(183, 175, 129, 0.05);--mantine-color-purple-text: var(--mantine-color-purple-4);--mantine-color-purple-filled: var(--mantine-color-purple-8);--mantine-color-purple-filled-hover: var(--mantine-color-purple-9);--mantine-color-purple-light: rgba(125, 62, 201, 0.15);--mantine-color-purple-light-hover: rgba(125, 62, 201, 0.2);--mantine-color-purple-light-color: var(--mantine-color-purple-3);--mantine-color-purple-outline: var(--mantine-color-purple-4);--mantine-color-purple-outline-hover: rgba(149, 98, 210, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-marble-6);--mantine-color-gray-light: rgba(97, 110, 165, 0.1);--mantine-color-gray-light-hover: rgba(97, 110, 165, 0.12);--mantine-color-gray-outline-hover: rgba(97, 110, 165, 0.05);--mantine-color-marble-text: var(--mantine-color-marble-filled);--mantine-color-marble-filled: var(--mantine-color-marble-6);--mantine-color-marble-filled-hover: var(--mantine-color-marble-7);--mantine-color-marble-light: rgba(167, 157, 98, 0.1);--mantine-color-marble-light-hover: rgba(167, 157, 98, 0.12);--mantine-color-marble-light-color: var(--mantine-color-marble-6);--mantine-color-marble-outline: var(--mantine-color-marble-6);--mantine-color-marble-outline-hover: rgba(167, 157, 98, 0.05);--mantine-color-purple-text: var(--mantine-color-purple-filled);--mantine-color-purple-filled: var(--mantine-color-purple-6);--mantine-color-purple-filled-hover: var(--mantine-color-purple-7);--mantine-color-purple-light: rgba(125, 62, 201, 0.1);--mantine-color-purple-light-hover: rgba(125, 62, 201, 0.12);--mantine-color-purple-light-color: var(--mantine-color-purple-6);--mantine-color-purple-outline: var(--mantine-color-purple-6);--mantine-color-purple-outline-hover: rgba(125, 62, 201, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <li
    class="_card_f5603a _opponentCard_f5603a m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
    style="--paper-radius: var(--mantine-radius-lg);"
  >
    <div>
      
          
      <p>
        Didn't melt fairer keepsakes since Fellowship elsewhere.
      </p>
      
          
      <p>
        Woodlands payment 
        <a
          href="#"
        >
          Osgiliath
        </a>
         tightening. Barad-dur follow belly comforts tender tough bell? Many that live deserve death. Some that die deserve life. Outwitted teatime grasp defeated before stones reflection corset seen animals Saruman's call?
      </p>
      
          
      <p>
        Curse you and all the halflings! Deserted anytime Lake-town burned caves balls. Smoked lthilien forbids Thrain? 
        <span
          style="font-size: 2.0rem"
        >
          Tad survive ensnare joy mistake courtesy Bagshot Row.
        </span>
         Ligulas step drops both? You shall not pass!
      </p>
      
          
      <p>
        Branch hey-diddle-diddle pony trouble'll sleeping during jump Narsil. 
        <span
          style="font-size: 0.6rem"
        >
          North valor overflowing sort Iáve mister kingly money?
        </span>
         Tender respectable success Valar impressive unfriendly bloom scraped?
      </p>
      
          
      <ul>
        
            
        <li>
          Adamant.
        </li>
        
            
        <li>
          Southfarthing!
        </li>
        
            
        <li>
          Witch-king.
        </li>
        
            
        <li>
          Precious.
        </li>
        
            
        <li>
          Gaffer's!
        </li>
        
          
      </ul>
      
          
      <ul>
        
            
        <li>
          Excuse tightening yet survives two cover Undómiel city ablaze.
        </li>
        
            
        <li>
          Keepsakes deeper clouds Buckland position 21 lied bicker fountains ashamed.
        </li>
        
            
        <li>
          Women rippling cold steps rules Thengel finer.
        </li>
        
            
        <li>
          Portents close Havens endured irons hundreds handle refused sister?
        </li>
        
            
        <li>
          Harbor Grubbs fellas riddles afar!
        </li>
        
          
      </ul>
      
          
      <p>
        Ravens wonder wanted runs me crawl gaining lots faster! Khazad-dum surprise baby season ranks. I bid you all a very fond farewell.
      </p>
      
          
      <ol>
        
            
        <li>
          Narsil.
        </li>
        
            
        <li>
          Elros.
        </li>
        
            
        <li>
          Arwen Evenstar.
        </li>
        
            
        <li>
          Maggot's?
        </li>
        
            
        <li>
          Bagginses?
        </li>
        
          
      </ol>
      
          
      <ol>
        
            
        <li>
          Concerning Hobbits l golf air fifth bell prolonging camp.
        </li>
        
            
        <li>
          Grond humble rods nearest mangler.
        </li>
        
            
        <li>
          Enormity Lórien merry gravy stayed move.
        </li>
        
            
        <li>
          Diversion almost notion furs between fierce laboring Nazgûl ceaselessly parent.
        </li>
        
            
        <li>
          Agree ruling um wasteland Bagshot Row expect sleep.
        </li>
        
          
      </ol>
      
          
      <p>
        Delay freezes Gollum. Let the Ring-bearer decide. Bagshot Row chokes pole pauses immediately orders taught éored musing three-day? Disease rune repel source fire Goblinses already?
      </p>
      
          
      <hr />
      
          
      <p>
        What about second breakfast? Nags runt near Lindir lock discover level? Andûril breathe waited flatten union.
      </p>
      
          
      <blockquote>
        
            
        <p>
          You shall be the Fellowship of the Ring.
        </p>
        
            
        <footer>
          —Númenor, 
          <cite>
            sweeter burned verse
          </cite>
        </footer>
        
          
      </blockquote>
      
          
      <p>
        I think we should get off the road. Penalty sight splintered Misty Mountain mithril? Unrest lasts rode league bears absence Bracegirdle athletic contract nice parent slowed?
      </p>
      
          
      <pre>
        Pardon Concerning Hobbits rune goblins? Twitching figure including rightful Thorin's level! Worth tubers threats Hornburg deadliest? Unfold thumping shh wants Homely!
      </pre>
      
          
      <hr />
    </div>
  </li>
</div>
`;

exports[`Turn component > Your turn > renders correctly 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: mocked, sans-serif;--mantine-font-family-headings: mocked, sans-serif;--mantine-primary-color-filled: var(--mantine-color-marble-filled);--mantine-primary-color-filled-hover: var(--mantine-color-marble-filled-hover);--mantine-primary-color-light: var(--mantine-color-marble-light);--mantine-primary-color-light-hover: var(--mantine-color-marble-light-hover);--mantine-primary-color-light-color: var(--mantine-color-marble-light-color);--mantine-primary-color-0: var(--mantine-color-marble-0);--mantine-primary-color-1: var(--mantine-color-marble-1);--mantine-primary-color-2: var(--mantine-color-marble-2);--mantine-primary-color-3: var(--mantine-color-marble-3);--mantine-primary-color-4: var(--mantine-color-marble-4);--mantine-primary-color-5: var(--mantine-color-marble-5);--mantine-primary-color-6: var(--mantine-color-marble-6);--mantine-primary-color-7: var(--mantine-color-marble-7);--mantine-primary-color-8: var(--mantine-color-marble-8);--mantine-primary-color-9: var(--mantine-color-marble-9);--mantine-color-gray-0: #f0f3fd;--mantine-color-gray-1: #e0e3ee;--mantine-color-gray-2: #bfc4d9;--mantine-color-gray-3: #9ca4c5;--mantine-color-gray-4: #7e88b3;--mantine-color-gray-5: #6b77a9;--mantine-color-gray-6: #616ea5;--mantine-color-gray-7: #515d91;--mantine-color-gray-8: #475283;--mantine-color-gray-9: #3a4775;--mantine-color-marble-0: #faf7ea;--mantine-color-marble-1: #efecde;--mantine-color-marble-2: #dbd7bf;--mantine-color-marble-3: #c8c29d;--mantine-color-marble-4: #b7af81;--mantine-color-marble-5: #aca36e;--mantine-color-marble-6: #a79d62;--mantine-color-marble-7: #918951;--mantine-color-marble-8: #817945;--mantine-color-marble-9: #706836;--mantine-color-purple-0: #f6eeff;--mantine-color-purple-1: #e7daf7;--mantine-color-purple-2: #cab1ea;--mantine-color-purple-3: #ad86dd;--mantine-color-purple-4: #9562d2;--mantine-color-purple-5: #854bcb;--mantine-color-purple-6: #7d3ec9;--mantine-color-purple-7: #6b31b2;--mantine-color-purple-8: #5f2aa0;--mantine-color-purple-9: #52228d;}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-marble-4);--mantine-color-gray-light: rgba(97, 110, 165, 0.15);--mantine-color-gray-light-hover: rgba(97, 110, 165, 0.2);--mantine-color-gray-outline-hover: rgba(126, 136, 179, 0.05);--mantine-color-marble-text: var(--mantine-color-marble-4);--mantine-color-marble-filled: var(--mantine-color-marble-8);--mantine-color-marble-filled-hover: var(--mantine-color-marble-9);--mantine-color-marble-light: rgba(167, 157, 98, 0.15);--mantine-color-marble-light-hover: rgba(167, 157, 98, 0.2);--mantine-color-marble-light-color: var(--mantine-color-marble-3);--mantine-color-marble-outline: var(--mantine-color-marble-4);--mantine-color-marble-outline-hover: rgba(183, 175, 129, 0.05);--mantine-color-purple-text: var(--mantine-color-purple-4);--mantine-color-purple-filled: var(--mantine-color-purple-8);--mantine-color-purple-filled-hover: var(--mantine-color-purple-9);--mantine-color-purple-light: rgba(125, 62, 201, 0.15);--mantine-color-purple-light-hover: rgba(125, 62, 201, 0.2);--mantine-color-purple-light-color: var(--mantine-color-purple-3);--mantine-color-purple-outline: var(--mantine-color-purple-4);--mantine-color-purple-outline-hover: rgba(149, 98, 210, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-marble-6);--mantine-color-gray-light: rgba(97, 110, 165, 0.1);--mantine-color-gray-light-hover: rgba(97, 110, 165, 0.12);--mantine-color-gray-outline-hover: rgba(97, 110, 165, 0.05);--mantine-color-marble-text: var(--mantine-color-marble-filled);--mantine-color-marble-filled: var(--mantine-color-marble-6);--mantine-color-marble-filled-hover: var(--mantine-color-marble-7);--mantine-color-marble-light: rgba(167, 157, 98, 0.1);--mantine-color-marble-light-hover: rgba(167, 157, 98, 0.12);--mantine-color-marble-light-color: var(--mantine-color-marble-6);--mantine-color-marble-outline: var(--mantine-color-marble-6);--mantine-color-marble-outline-hover: rgba(167, 157, 98, 0.05);--mantine-color-purple-text: var(--mantine-color-purple-filled);--mantine-color-purple-filled: var(--mantine-color-purple-6);--mantine-color-purple-filled-hover: var(--mantine-color-purple-7);--mantine-color-purple-light: rgba(125, 62, 201, 0.1);--mantine-color-purple-light-hover: rgba(125, 62, 201, 0.12);--mantine-color-purple-light-color: var(--mantine-color-purple-6);--mantine-color-purple-outline: var(--mantine-color-purple-6);--mantine-color-purple-outline-hover: rgba(125, 62, 201, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <li
    class="_card_f5603a m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
    style="--paper-radius: var(--mantine-radius-lg);"
  >
    <div>
      
          
      <p>
        Didn't melt fairer keepsakes since Fellowship elsewhere.
      </p>
      
          
      <p>
        Woodlands payment 
        <a
          href="#"
        >
          Osgiliath
        </a>
         tightening. Barad-dur follow belly comforts tender tough bell? Many that live deserve death. Some that die deserve life. Outwitted teatime grasp defeated before stones reflection corset seen animals Saruman's call?
      </p>
      
          
      <p>
        Curse you and all the halflings! Deserted anytime Lake-town burned caves balls. Smoked lthilien forbids Thrain? 
        <span
          style="font-size: 2.0rem"
        >
          Tad survive ensnare joy mistake courtesy Bagshot Row.
        </span>
         Ligulas step drops both? You shall not pass!
      </p>
      
          
      <p>
        Branch hey-diddle-diddle pony trouble'll sleeping during jump Narsil. 
        <span
          style="font-size: 0.6rem"
        >
          North valor overflowing sort Iáve mister kingly money?
        </span>
         Tender respectable success Valar impressive unfriendly bloom scraped?
      </p>
      
          
      <ul>
        
            
        <li>
          Adamant.
        </li>
        
            
        <li>
          Southfarthing!
        </li>
        
            
        <li>
          Witch-king.
        </li>
        
            
        <li>
          Precious.
        </li>
        
            
        <li>
          Gaffer's!
        </li>
        
          
      </ul>
      
          
      <ul>
        
            
        <li>
          Excuse tightening yet survives two cover Undómiel city ablaze.
        </li>
        
            
        <li>
          Keepsakes deeper clouds Buckland position 21 lied bicker fountains ashamed.
        </li>
        
            
        <li>
          Women rippling cold steps rules Thengel finer.
        </li>
        
            
        <li>
          Portents close Havens endured irons hundreds handle refused sister?
        </li>
        
            
        <li>
          Harbor Grubbs fellas riddles afar!
        </li>
        
          
      </ul>
      
          
      <p>
        Ravens wonder wanted runs me crawl gaining lots faster! Khazad-dum surprise baby season ranks. I bid you all a very fond farewell.
      </p>
      
          
      <ol>
        
            
        <li>
          Narsil.
        </li>
        
            
        <li>
          Elros.
        </li>
        
            
        <li>
          Arwen Evenstar.
        </li>
        
            
        <li>
          Maggot's?
        </li>
        
            
        <li>
          Bagginses?
        </li>
        
          
      </ol>
      
          
      <ol>
        
            
        <li>
          Concerning Hobbits l golf air fifth bell prolonging camp.
        </li>
        
            
        <li>
          Grond humble rods nearest mangler.
        </li>
        
            
        <li>
          Enormity Lórien merry gravy stayed move.
        </li>
        
            
        <li>
          Diversion almost notion furs between fierce laboring Nazgûl ceaselessly parent.
        </li>
        
            
        <li>
          Agree ruling um wasteland Bagshot Row expect sleep.
        </li>
        
          
      </ol>
      
          
      <p>
        Delay freezes Gollum. Let the Ring-bearer decide. Bagshot Row chokes pole pauses immediately orders taught éored musing three-day? Disease rune repel source fire Goblinses already?
      </p>
      
          
      <hr />
      
          
      <p>
        What about second breakfast? Nags runt near Lindir lock discover level? Andûril breathe waited flatten union.
      </p>
      
          
      <blockquote>
        
            
        <p>
          You shall be the Fellowship of the Ring.
        </p>
        
            
        <footer>
          —Númenor, 
          <cite>
            sweeter burned verse
          </cite>
        </footer>
        
          
      </blockquote>
      
          
      <p>
        I think we should get off the road. Penalty sight splintered Misty Mountain mithril? Unrest lasts rode league bears absence Bracegirdle athletic contract nice parent slowed?
      </p>
      
          
      <pre>
        Pardon Concerning Hobbits rune goblins? Twitching figure including rightful Thorin's level! Worth tubers threats Hornburg deadliest? Unfold thumping shh wants Homely!
      </pre>
      
          
      <hr />
    </div>
  </li>
</div>
`;
